MSAL/src/requests/MSALBaseRequest.m

```diff
@@ -187,113 +208,52 @@ - (void)acquireToken:(nonnull MSALCompletionBlock)completionBlock
+             completionBlock(nil, [MSALErrorConverter MSALErrorFromMSIDError:error]);
              return;
          }
          
-         NSString *oauthError = tokenResponse.error;
-         if (oauthError)
+         if (![factory verifyResponse:tokenResponse context:nil error:&error])
          {
-             MSALErrorCode code = MSALErrorCodeForOAuthError(oauthError, MSALErrorInteractionRequired);
-             
-             NSError *msalError = CREATE_MSID_LOG_ERROR_WITH_SUBERRORS(_parameters, code, oauthError, tokenResponse.subError, @"%@", tokenResponse.errorDescription);
-             
-             [self stopTelemetryEvent:event error:msalError];
-             
-             completionBlock(nil, msalError);
+             [self stopTelemetryEvent:event error:error];
              
+             completionBlock(nil, [MSALErrorConverter MSALErrorFromMSIDError:error]);
              return;
          }
-         
-         if ([NSString msidIsStringNilOrBlank:tokenResponse.scope])
-         {
-             MSID_LOG_INFO(_parameters, @"No scope in server response, using passed in scope instead.");
-             MSID_LOG_INFO_PII(_parameters, @"No scope in server response, using passed in scope instead.");
-             tokenResponse.scope = _parameters.scopes.msalToString;
-         }
-         
-         // For silent flow, with grant type being MSID_OAUTH2_REFRESH_TOKEN, this value may be missing from the response.
-         // In this case, we simply return the refresh token in the request.
-         if ([reqParameters[MSID_OAUTH2_GRANT_TYPE] isEqualToString:MSID_OAUTH2_REFRESH_TOKEN])
-         {
-             if (!tokenResponse.refreshToken)
-             {
-                 tokenResponse.refreshToken = reqParameters[MSID_OAUTH2_REFRESH_TOKEN];
-                 MSID_LOG_WARN(_parameters, @"Refresh token was missing from the token refresh response, so the refresh token in the request is returned instead");
-                 MSID_LOG_WARN_PII(_parameters, @"Refresh token was missing from the token refresh response, so the refresh token in the request is returned instead");
-             }
-         }
-         
-         // Check user mismatch
-         MSALClientInfo *clientInfo = [[MSALClientInfo  alloc] initWithRawClientInfo:tokenResponse.clientInfo
-                                                                               error:&error];
-         if (!clientInfo)
-         {
-             MSID_LOG_ERROR(_parameters, @"Client info was not returned in the server response");
-             MSID_LOG_ERROR_PII(_parameters, @"Client info was not returned in the server response");
-             completionBlock(nil, error);
-             return;
-         }
+
          if (_parameters.user != nil &&
-             ![_parameters.user.userIdentifier isEqualToString:clientInfo.userIdentifier])
+             ![_parameters.user.userIdentifier isEqualToString:tokenResponse.clientInfo.userIdentifier])
```

Comment:
I feel like userIdentifier check has to be somewhere else, but we can leave that change for future